.github/workflows/dockerCI.yaml

```diff
@@ -24,50 +22,58 @@ jobs:
 
     runs-on: ubuntu-latest
     env:
-      # copy the secrets to env vars for if: conditions
+      # copy the config value to environment variables
+      # do not copy the actual secrets!
 
+      KEYVAULT_NAME: ${{ secrets.KEYVAULT_NAME }}
       DOCKER_REPO: ${{ secrets.DOCKER_REPO }}
-      DOCKER_USER: ${{ secrets.DOCKER_USER }}
-      DOCKER_PAT: ${{ secrets.DOCKER_PAT }}
-
-      TENANT: ${{ secrets.TENANT }}
-      SERVICE_PRINCIPAL: ${{ secrets.SERVICE_PRINCIPAL }}
-      SERVICE_PRINCIPAL_SECRET: ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
-
-      ACR_REG: ${{ secrets.ACR_REG }}
       ACR_REPO: ${{ secrets.ACR_REPO }}
       ACR_IMAGE: ${{ secrets.ACR_IMAGE }}
+      ACR_REG: ${{ secrets.ACR_REG }}
 
-      KEYVAULT_NAME: ${{ secrets.KEYVAULT_NAME }}
-      AUTH_TYPE: CLI
-
+      # These are used in if: conditions
       AZURE_SET: false
       DOCKER_SET: false
       ACR_SET: false
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
 
     - name: Set conditions
       run: |
-        echo "::set-env name=AZURE_SET::${{ env.SERVICE_PRINCIPAL_SECRET != '' && env.TENANT != '' && env.SERVICE_PRINCIPAL != '' && env.KEYVAULT_NAME != '' }}"
-        echo "::set-env name=DOCKER_SET::${{ env.DOCKER_PAT != '' && env.DOCKER_REPO != '' && env.DOCKER_USER != '' }}"
-        echo "::set-env name=ACR_SET::${{ env.ACR_REG != '' && env.ACR_REPO != '' && env.ACR_IMAGE !='' }}"
+        # check that all Azure secrets are set
+        if [[ "${{ secrets.SERVICE_PRINCIPAL_SECRET }}" != "" && "${{ secrets.TENANT }}" != "" && "${{ secrets.SERVICE_PRINCIPAL }}" != "" && "${{ secrets.KEYVAULT_NAME }}" != "" ]];
+        then
+          echo "::set-env name=AZURE_SET::true"
+        fi
+
+        # check that all Docker secrets are set
+        if [[ "${{ secrets.DOCKER_PAT }}" != "" && "${{ secrets.DOCKER_REPO }}" != "" && "${{ secrets.DOCKER_USER }}" != "" ]];
+        then
+          echo "::set-env name=DOCKER_SET::true"
+        fi
+        
+        # check that all ACR secrets are set
+        if [[ "${{ secrets.ACR_REG }}" != "" && "${{ secrets.ACR_REPO }}" != "" && "${{ secrets.ACR_IMAGE }}" != "" ]];
+        then
+          echo "ACR Set"
+          echo "::set-env name=ACR_SET::true"
+        fi
 
     - name: Validate Azure Access
       if: ${{ env.AZURE_SET == 'true' }}
       run: |
         # login to Azure
-        az login --service-principal -u ${SERVICE_PRINCIPAL} --tenant ${TENANT} -p ${SERVICE_PRINCIPAL_SECRET}
+        az login --service-principal -u ${{ secrets.SERVICE_PRINCIPAL }} --tenant ${{ secrets.TENANT }} -p ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
 
```

Comment:
@bartr : Lines 63 - 172 have a number of if statements and seem dependent successful completion on a previous block.  Should these be refactored to have a job identifiers and then the dependent block can declare needs.[job id].result.  Then if the conditional is false it could return a result of false making the dependency evaluate to false and the subsequent block won't execute?  Disclaimer: I'm likely the least experienced in yaml on the team.

Reply:
Interesting idea ... I looked into creating a custom task via java script, but you don't get the secrets. I didn't think about packaging it as a step in yaml and using the result instead of the env var. That's definitely worth looking into and could be more elegant.